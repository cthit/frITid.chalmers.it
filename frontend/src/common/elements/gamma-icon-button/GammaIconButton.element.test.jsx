import React from "react";
import { shallow } from "enzyme";
import GammaIconButton from "./";
import { Add, MoreVert } from "@material-ui/icons";

describe("<GammaIconButton/>", () => {
  test("Shallow render of a primary <GammaIconButton/> with a Add icon", () => {
    const wrapper = shallow(
      <GammaIconButton primary component={Add} onClick={() => {}} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("Shallow render of a secondary disabled<GammaIconButton/> with a MoreVert icon", () => {
    const wrapper = shallow(
      <GammaIconButton
        secondary
        disabled
        component={MoreVert}
        onClick={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
