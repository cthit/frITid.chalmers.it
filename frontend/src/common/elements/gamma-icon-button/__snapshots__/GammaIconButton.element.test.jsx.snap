// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GammaIconButton/> Shallow render of a primary <GammaIconButton/> with a Add icon 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GammaIconButton
    component={[Function]}
    onClick={[Function]}
    primary={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <pure(Add) />,
      "color": "primary",
      "disabled": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <pure(Add) />,
        "color": "primary",
        "disabled": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<GammaIconButton/> Shallow render of a secondary disabled<GammaIconButton/> with a MoreVert icon 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GammaIconButton
    component={[Function]}
    disabled={true}
    onClick={[Function]}
    secondary={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <pure(MoreVert) />,
      "color": "secondary",
      "disabled": true,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <pure(MoreVert) />,
        "color": "secondary",
        "disabled": true,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
